pilot/pkg/xds/pcds.go

```diff
@@ -30,6 +31,10 @@ type PcdsGenerator struct {
 var _ model.XdsResourceGenerator = &PcdsGenerator{}
 
 func pcdsNeedsPush(req *model.PushRequest) bool {
+	if !features.MultiRootMesh.Get() {
```

Comment:
This just means that we will not push updates. But the proxies will still request PCDS and we will still return something.

I think we also need something at the proxy level to tell it to not request PCDS at all?

Reply:
wasn't sure if we needed that.., but will add this change as well

Reply:
on second thought, it does look like the pull from the proxy is fairly harmless. It specifically fires off a request for [TypeUrl: v3.ProxyConfigType]. if Istiod PCDS generator doesn't respond to the request, is there a concern that other generators will?

Reply:
If we do want the feature flag to be consumed by the istio-agent/gateways, any suggestions on whether we should include this feature capability as a new field in MeshConfig? Or should we just be able to configure an environment variable directly through istioctl into istiod have istiod pass the feature capability flag into Istio-proxies/gateways? Is this being done for any other features?

Reply:
Yeah we have plenty of env vars on the agent, PROXY_XDS_VIA_AGENT is an example of a similar one where its just a feature flag.